silo-vaults/test/foundry/MetaMorphoFactoryTest.sol

```diff
@@ -31,28 +31,30 @@ contract MetaMorphoFactoryTest is IntegrationTest {
         vm.assume(address(initialOwner) != address(0));
         initialTimelock = bound(initialTimelock, ConstantsLib.MIN_TIMELOCK, ConstantsLib.MAX_TIMELOCK);
 
-        bytes32 initCodeHash = hashInitCode(
-            type(MetaMorpho).creationCode,
-            abi.encode(initialOwner, initialTimelock, address(loanToken), name, symbol)
-        );
-        address expectedAddress = computeCreate2Address(salt, initCodeHash, address(factory));
-
-        vm.expectEmit(address(factory));
+        vm.expectEmit(false, false, false, false);
         emit EventsLib.CreateMetaMorpho(
-            expectedAddress, address(this), initialOwner, initialTimelock, address(loanToken), name, symbol, salt
+            address(0), address(this), initialOwner, initialTimelock, address(loanToken), name, symbol, salt
         );
 
         IMetaMorpho metaMorpho =
             factory.createMetaMorpho(initialOwner, initialTimelock, address(loanToken), name, symbol, salt);
 
-        assertEq(expectedAddress, address(metaMorpho), "computeCreate2Address");
-
         assertTrue(factory.isMetaMorpho(address(metaMorpho)), "isMetaMorpho");
 
         assertEq(metaMorpho.owner(), initialOwner, "owner");
         assertEq(metaMorpho.timelock(), initialTimelock, "timelock");
         assertEq(metaMorpho.asset(), address(loanToken), "asset");
         assertEq(metaMorpho.name(), name, "name");
         assertEq(metaMorpho.symbol(), symbol, "symbol");
+        assertTrue(address(metaMorpho.INCENTIVES_MODULE()) != address(0), "INCENTIVES_MODULE");
+
+        // we can still generate correct address but we have to predict INCENTIVES_MODULE address to do so
```

Comment:
[cloneDeterministic](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/proxy/Clones.sol#L74) and [predictDeterministicAddress](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/proxy/Clones.sol#L98C14-L98C41) will solve the issue with address prediction. We use it in the SiloDeployer, see [here](https://github.com/silo-finance/silo-contracts-v2/blob/develop/silo-core/contracts/SiloDeployer.sol#L90) ([lib](https://github.com/silo-finance/silo-contracts-v2/blob/develop/silo-core/contracts/lib/CloneDeterministic.sol))

Reply:
so, you think this is necessary feature here?

Reply:
>TODO: because of cloning (in factory), it will be harder to predict address of vault, should we remove cloning?

The proposed solution makes it easy.

Reply:
> so, you think this is necessary feature here?

Probably no. I would also not clone a SiloIncentivesController to make the execution cheaper.

Reply:
Btw, let's deploy a SiloIncentivesController too.

Reply:
- deterministic clonning done
- SiloIncentivesController - I will create separate PR